packages/shared/src/util/index.ts

```diff
@@ -176,6 +208,8 @@ export function getMatchingRules(
       ([environmentId, settings]) => {
         if (!isValidEnvironment(environmentId, environments)) return;
 
+        if (omitDisabledEnvironments && !settings.enabled) return;
```

Comment:
@jdorn This is the risky part I mentioned. It feels like if an environment's feature-level toggle is disabled, we don't want to include the environment, even if the environment has an enabled ff on it.